docs/admin/resource_constraints.md

```diff
@@ -0,0 +1,182 @@
+<!--[metadata]>
++++
+aliases = ["/engine/articles/systemd/"]
+title = "Constrain a container's resources"
+description = "Limiting the system resources a container can use"
+keywords = ["docker, daemon, configuration"]
+[menu.main]
+parent = "engine_admin"
+weight="7"
++++
+<![end-metadata]-->
+
+# Limiting a container's access to system resources
+
+By default, a container has no resource constraints and can use as much of a
+given system resource as the host's kernel scheduler will allow. Docker provides
+ways to control control how much memory, CPU, or block IO a container can use,
+either at image creation or at runtime by setting configuration flags of the
+`docker run` command. This section provides details on when you should set such
+limits and the possible implications of setting them.
+
+>**Note**: It is not possible to limit a container's network bandwidth at this time.
+This is being tracked in issue []#4763](https://github.com/docker/docker/issues/4763).
+
+## memory
+
+Docker can enforce hard memory limits, which allow the container to use no more
+than a given amount of user or system memory, or soft limits, which allow the
+container to use as much memory as it needs unless certain conditions are met,
+such as when Docker detects low memory or contention on the host machine. Some of
+these options have different effects when used alone or when more than one options
+is set.
+
+Most of these options take a positive integer, followed by a suffix of `b`, `k`,
+`m`, `g`, to indicate the unit.
+
+- **`-m` or `--memory=**: The maximum amount of memory the container can use.
+  If you set this option, the minimum allowed value is `4m`.
+
+- **`--memory-swap`**: The amount of memory this container is allowed to swap to
+  disk.
+  <ul>
+    <li>If unset, and `--memory` is set, the container can use twice as much swap
+        as the `--memory` setting, if the host container has swap memory configured.
+        For instance, if `--memory="300m"` and `--memory-swap` is not set, the
+        container can use 300m of memory and 600m of swap.</li>
+    <li>If set to a positive integer,  and if both `--memory` and `--memory-swap`
+        are set, `--memory-swap` represents the total amount of memory and swap
+        that can be used, and `--memory` controls the amount used by non-swap
+        memory. So if `--memory="300m"` and `--memory-swap="1g"`, the container
+        can use 300m of memory and 700m (1g - 300m) swap.</li>
+    <li>If set to `-1`, the container is allowed to use unlimited swap memory.</li>
+  </ul>
+
+- **`--memory-swappiness`**: By default, a container's kernel can swap out a
+  percentage of anonymous pages. You can set `--memory-swappiness` to a value
+  between 0 and 100, to tune this percentage. <ul><li>A value of 0 turns off anonymous
```

Comment:
I'd prefer a markdown bullet-list for this, but let me know if that complicated things (just indenting more _should_ create a nested bullet-list in markdown)